### PR TITLE
pull: Fix pull --bottle 1234

### PR DESCRIPTION
**Pull request by sjackman**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Fix the error:
```
Error: undefined method `casecmp' for nil:NilClass
```

### CODE REVIEW

**Comment by sjackman** [comment]
This PR fixes a bug that I introduced in PR #3279. Sorry for that.

**Comment by MikeMcQuaid** [comment]
@sjackman This is failing on the Linux CI. Note the Linux CI should not end up cloning Linuxbrew in any situation.

**Comment by scpeters** [comment]
> Note the Linux CI should not end up cloning Linuxbrew in any situation.

I see references to linuxbrew repositories in the following places:

https://github.com/Homebrew/brew/blob/ffb582b5a6d829c5825e6d6a40545cb3099001fb/Library/Homebrew/cmd/update.sh#L32-L33

https://github.com/Homebrew/brew/blob/3a606abaa13a3090f3c6336efed435006f091c60/Library/Homebrew/tap.rb#L544

**Comment by MikeMcQuaid** [comment]
@scpeters Yep, those are fine but we've previously taken lengths to workaround that in CI itself.

**Comment by JCount** [comment]
@sjackman 
The CI issue seems caused by the use of [CoreTap's default_remote method](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/tap.rb#L539-L546),  due to its platform-specific behavior. This is likely cause for the test suite to cloning `https://github.com/Linuxbrew/homebrew-core` when it runs `brew pull 1` as part of the [pull_spec](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/dev-cmd/pull_spec.rb#L20-L23) tests. I suspect that
```
arg = "#{CoreTap.instance.default_remote}/pull/#{arg}" if arg.to_i.positive?
```
 is evaluating to 
```
arg = "https://github.com/Linuxbrew/homebrew-core/pull/1"
```
on the Linux CI.

**Comment by sjackman** [comment]
I've fixed the test failure. I've removed the synonym of the tap name `Homebrew/homebrew` for `Homebrew/core`. I can put it back in if you prefer.

File: Library/Homebrew/tap.rb [hunk 1]
```diff
@@ -27,10 +27,10 @@ def self.fetch(*args)
     end
 
     # we special case homebrew so users don't have to shift in a terminal
-    user = "Homebrew" if user == "homebrew"
+    user = user.capitalize if ["homebrew", "linuxbrew"].include? user
     repo = repo.strip_prefix "homebrew-"
 
-    if user == "Homebrew" && (repo == "homebrew" || repo == "core")
```
Comment (MikeMcQuaid) [inline Library/Homebrew/tap.rb]: Does the `repo == "homebrew"` break anything here? If it's not required for the fix can you put it back? Thanks!

File: Library/Homebrew/test/dev-cmd/pull_spec.rb [hunk 2]
```diff
@@ -22,7 +22,7 @@
       .and output(/Current branch is new\-branch/).to_stderr
       .and be_a_failure
 
-    expect { brew "pull", "--bump", "8" }
+    expect { brew "pull", "--bump", "https://github.com/Homebrew/homebrew-core/pull/8" }
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/dev-cmd/pull_spec.rb]: To be explicit: this is just to fix the test failure on Linux, right? Using `8` will still pull Homebrew/homebrew-core on macOS?

File: Library/Homebrew/tap.rb [hunk 3]
```diff
@@ -27,10 +27,10 @@ def self.fetch(*args)
     end
 
     # we special case homebrew so users don't have to shift in a terminal
```
Comment (MikeMcQuaid) [inline Library/Homebrew/tap.rb]: `and linuxbrew`

File: Library/Homebrew/tap.rb (hunk 1 above)
Comment (sjackman) [reply Library/Homebrew/tap.rb]: It doesn't break anything. Just seemed to be an anachronism. I can put it back.

File: Library/Homebrew/test/dev-cmd/pull_spec.rb (hunk 2 above)
Comment (sjackman) [reply Library/Homebrew/test/dev-cmd/pull_spec.rb]: Correct.

**Comment by MikeMcQuaid** [comment]
Thanks again @sjackman and sorry for all the back and forth.

**Comment by sjackman** [comment]
No worries. Thanks for merging, Mike.

**Pull request merged**